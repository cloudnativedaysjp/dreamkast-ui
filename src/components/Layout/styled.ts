import styled from 'styled-components'

import Toolbar from '@material-ui/core/Toolbar'

export const Container = styled.div<{ eventAbbr?: string }>`
  background-image: url('/${(props) => props.eventAbbr}/ui/background.jpg');
  background-position: 50%;
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  min-height: 100vh;
`

export const Header = styled(Toolbar)`
  display: flex;
  justify-content: flex-start;
  background-color: #ffffff;
  color: #282828;
`

export const HeaderImg = styled.img`
  height: 40px;
`

export const ChildrenContainer = styled.div`
  background: rgba(255, 255, 255, 0.6);
  padding-bottom: 22px;
`

export const Footer = styled.footer`
  grid-row-start: 2;
  grid-row-end: 3;
  background-color: #fff;
  width: 100%;
  height: 6vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`

export const FooterText = styled.div`
  font-size: 14px;
  color: gray;
`
export const FooterLink = styled.a`
  text-decoration: none;
  font-size: 14px;
  color: gray;
`

export const MenuLink = styled.a`
  text-decoration-line: none;
`

export const DesktopMenu = styled.div`
  display: none;
  justify-content: flex-end;
  margin-left: auto;
  @media (min-width: 600px) {
    display: flex;
  }
`

export const MobileMenu = styled.div`
  display: none;
  justify-content: flex-end;
  margin-left: auto;
  @media (max-width: 599px) {
    display: flex;
  }
`
