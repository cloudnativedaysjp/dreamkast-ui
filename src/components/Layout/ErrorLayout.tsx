import React from 'react'
import * as Styled from './styled'
import Head from 'next/head'
import AppBar from '@material-ui/core/AppBar'
import Link from 'next/link'
import { useRouter } from 'next/router'

type Props = {
  statusCode?: number
}

export const ErrorLayout = ({ statusCode }: Props) => {
  const router = useRouter()
  const eventAbbr = (() => {
    const { eventAbbr } = router.query
    return router.isReady ? (eventAbbr as string) : ''
  })()

  return (
    <Styled.Container eventAbbr={eventAbbr}>
      <Head>
        <title>Error: CloudNative Days player</title>
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      <header>
        <AppBar position="static">
          <Styled.Header>
            <Link href={`/${eventAbbr}/ui`} rel="noopener noreferrer">
              <Styled.HeaderImg
                src={`/${eventAbbr}/ui/${eventAbbr}_header_logo.png`}
              />
            </Link>
          </Styled.Header>
        </AppBar>
      </header>
      <Styled.ErrorContainer>
        <Styled.ErrorMessage>
          {statusCode || ''} エラーが発生しました。
          <br />
          リロードしても問題が継続する場合は、スタッフまでお問い合わせください。
        </Styled.ErrorMessage>
      </Styled.ErrorContainer>
    </Styled.Container>
  )
}
