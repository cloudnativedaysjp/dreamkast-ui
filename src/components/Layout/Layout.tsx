import React, { ReactNode, useEffect, useState } from 'react'
import * as Styled from './styled'
import Head from 'next/head'
import { DesktopMenu, MobileMenu } from '../Menu'
import AppBar from '@material-ui/core/AppBar'
import { Event } from '../../client-axios'

type Props = {
  children?: ReactNode
  title?: string
  event?: Event
}

export const Layout: React.FC<Props> = ({
  children,
  title = 'This is the default title',
  event,
}) => {
  const [url, setUrl] = useState<string>('')

  useEffect(() => {
    if (typeof window !== 'undefined') {
      setUrl(window.location.origin + '/logout')
    }
  }, [])

  return (
    <Styled.Container eventAbbr={event?.abbr}>
      <Head>
        <title>{title}</title>
        <link
          rel="shortcut icon"
          href={`/${event?.abbr}/ui/${event?.abbr}_trademark.png`}
        />
        <meta charSet="utf-8" />
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      <header>
        <AppBar position="static">
          <Styled.Header>
            <a href={`/${event?.abbr}/ui`} rel="noopener noreferrer">
              <Styled.HeaderImg
                src={`/${event?.abbr}/ui/${event?.abbr}_header_logo.png`}
              />
            </a>
            <DesktopMenu event={event} url={url} />
          </Styled.Header>
          <MobileMenu event={event} url={url} />
        </AppBar>
      </header>
      <Styled.ChildrenContainer>{children}</Styled.ChildrenContainer>
      <Styled.Footer>
        <Styled.FooterLink
          href={`https://event.cloudnativedays.jp/${event?.abbr}/privacy`}
          target="_blank"
          rel="noopener noreferrer"
        >
          プライバシーポリシー
        </Styled.FooterLink>
        <Styled.FooterLink
          href={`https://event.cloudnativedays.jp/${event?.abbr}/coc`}
          target="_blank"
          rel="noopener noreferrer"
        >
          行動規範
        </Styled.FooterLink>
        <Styled.FooterText>{event?.copyright}</Styled.FooterText>
      </Styled.Footer>
    </Styled.Container>
  )
}
