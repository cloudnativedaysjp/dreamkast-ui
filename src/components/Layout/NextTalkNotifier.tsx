import React, { PropsWithChildren, useEffect } from 'react'
import {
  nextRegisteredTalkSelector,
  setInitialViewTalk,
  setViewTrackId,
} from '../../store/settings'
import { useDispatch, useSelector } from 'react-redux'
import {
  closeSnackbar,
  enqueueSnackbar,
  SnackbarKey,
  SnackbarProvider,
} from 'notistack'
import dayjs from 'dayjs'
import { Button } from '@material-ui/core'
import styled from 'styled-components'

export const NextTalkNotifier = ({ children }: PropsWithChildren) => {
  const dispatch = useDispatch()
  const { talk, track } = useSelector(nextRegisteredTalkSelector)

  const handleTrackChange = (trackId: number) => {
    dispatch(setViewTrackId(trackId))
    dispatch(setInitialViewTalk())
  }

  useEffect(() => {
    if (!talk || !track) {
      return
    }
    console.warn(talk)

    const component = (
      <div>
        <p>事前登録しているセッションが開始されました。</p>
        <p>
          {`トラック${track.name}: `}
          {dayjs(talk.startTime).tz().format('HH:mm')}-
          {dayjs(talk.endTime).tz().format('HH:mm')}
          <br />
          {talk.title}
          <br />
          {talk.speakers?.map((s) => s.name).join(', ')}
        </p>
      </div>
    )

    const action = (snackbarId: SnackbarKey) => (
      <div>
        <div>
          <ActionButton
            variant="contained"
            size="small"
            onClick={() => {
              handleTrackChange(track.id)
              closeSnackbar(snackbarId)
            }}
          >
            視聴する
          </ActionButton>
        </div>
        <div>
          <ActionButton
            variant="contained"
            size="small"
            onClick={() => {
              closeSnackbar(snackbarId)
            }}
          >
            閉じる
          </ActionButton>
        </div>
      </div>
    )

    enqueueSnackbar(component, { action })
  }, [talk])

  return (
    <SnackbarProvider maxSnack={6} autoHideDuration={10000} preventDuplicate>
      {children}
    </SnackbarProvider>
  )
}

const ActionButton = styled(Button)`
  width: 90px;
  margin-right: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
`
