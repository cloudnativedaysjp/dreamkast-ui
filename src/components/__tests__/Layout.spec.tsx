import React from 'react'
import renderer from 'react-test-renderer'
import Layout from '../Layout'

test('Layout', () => {
  const component = renderer.create(<Layout title="Test" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
