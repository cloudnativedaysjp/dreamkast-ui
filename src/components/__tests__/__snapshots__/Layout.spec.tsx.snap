// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
<div>
  <header>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          New Dreamkast
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
      </div>
    </header>
  </header>
  <footer>
    <hr />
    <span>
      CloudNative Days
    </span>
  </footer>
</div>
`;
